### PR TITLE
[review] Add etcd 3.1.12 to conda-forge

### PR DESCRIPTION
**Pull request by sodre**
- etcd is a distributed reliable key-value store for the most critical data of a distributed system
  - https://github.com/coreos/etcd

### CODE REVIEW

File: recipes/etcd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - git  # [not win]
+    - make  # [not win]
+    - toolchain  # [not win]
+
+test:
+  commands:
+    - etcd --version
+    - etcdctl --version
+
+    # Ugly hack because etcd&etcdclt are not a dynamic executable
```
Comment (sodre) [inline recipes/etcd/meta.yaml]: The hack is to skip linkages test in Linux.

**Comment by sodre** [comment]
@ocefpaf It looks like I caught at a good time. This is ready for review. 1/3

File: recipes/etcd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/coreos/{{ name }}/archive/v{{ version }}.tar.gz  # [not win]
+  url: https://github.com/coreos/{{ name }}/releases/download/v{{ version }}/{{ name }}-v{{ version }}-windows-amd64.zip  # [win]
+  fn: {{ name }}-{{ version }}.tar.gz  # [not win]
```
Comment (ocefpaf) [inline recipes/etcd/meta.yaml]: You can remove this entirely and let the url define the `fn` automatically.

File: recipes/etcd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - toolchain  # [not win]
+
+test:
+  commands:
+    - etcd --version
+    - etcdctl --version
+
+    # Ugly hack because etcd&etcdclt are not a dynamic executable
+    - conda inspect linkages -p $PREFIX {{ name|lower }}  # [osx]
+    - conda inspect objects -p $PREFIX {{ name|lower }}  # [osx]
```
Comment (ocefpaf) [inline recipes/etcd/meta.yaml]: For styling reasons we are using `$PKG_NAME` instead of `{{ name|lower }}`.

File: recipes/etcd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - make  # [not win]
+    - toolchain  # [not win]
+
+test:
+  commands:
+    - etcd --version
+    - etcdctl --version
+
+    # Ugly hack because etcd&etcdclt are not a dynamic executable
+    - conda inspect linkages -p $PREFIX {{ name|lower }}  # [osx]
```
Comment (ocefpaf) [inline recipes/etcd/meta.yaml]: This can be `# [not win]`

**Pull request merged**